presto-spi/src/main/java/io/prestosql/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -655,6 +610,7 @@ public static void multiply(Slice left, int right, Slice result)
         pack(result, (int) z0, (int) z1, (int) z2, (int) z3, leftNegative != rightNegative);
     }
 
+    // Using multiply(long, long, long, long, Slice) here decreases performance by ~40%
```

Comment:
This is not a TODO. The code is optimal (to some degree) at this point. The comment is made so that no one wants to try generalize it more and end up loosing performance.

Reply:
Consider rewriting this to stress the actual goal.